packages/core/src/html/scriptAndStyleTagProcessor.js

```diff
@@ -1,44 +1,30 @@
-/**
- * Adds attribute "type=application/javascript" to <script> tag.
- *
- * This is necessary because we are pre-compiling each page as a Vue application,
- * where the Vue compilation ignores and discards the <script> tag.
- *
- * By having this attribute, the compilation will ignore the <script> tag but not discard it.
- *
- * @param {Object<any, any>} node from the dom traversal
- */
-function processScriptTag(node) {
-  node.attribs.type = 'application/javascript';
-}
+const cheerio = require('cheerio');
 
 /**
- * Changes <style> tag into placeholder tag:
- * <script src defer type="application/javascript" style-bypass-vue-compilation>.
+ * Removes every script and style node (written by the user) encountered in the main app,
+ * and records all these script nodes in an array so that they can be hoisted to after
+ * <body> tag at a later stage.
  *
- * This is necessary because we are pre-compiling each page as a Vue application,
- * where the Vue compilation ignores and discards the <style> tag.
- *
- * We work around this problem by piggybacking on the approach we used for <script> tags to be ignored
- * and not discarded; by changing <style> tags to the placeholder tag as shown above.
- * We will change it back to <style> tag after Vue compilation and after the element is created by Vue.
+ * This is to prevent Vue compilation of script/style tags (as they are not meant to be compiled).
  *
  * @param {Object<any, any>} node from the dom traversal
  */
-function processStyleTag(node) {
-  node.name = 'script';
-  /*
-    This node is not intended to be ran as a script. Thus, we have to defer the execution of the script
-    until the node is restored as a style node (which is handled by MarkBind's script).
-    The defer attribute requires the src attribute to work. Thus, the src attribute is used as a dummy.
-  */
-  node.attribs.src = '';
-  node.attribs.defer = '';
-  node.attribs.type = 'application/javascript'; // to bypass Vue compilation as a script tag
-  node.attribs['style-bypass-vue-compilation'] = ''; // to allow specific query selection of this element
+function processScriptAndStyleTag(node, userScriptsAndStyles) {
+  // Do not process script tags that are meant to be inserted in head/bottom of HTML
+  const isHeadOrBottomScript = node.parent.name === 'head-top'
+    || node.parent.name === 'head-bottom' || node.parent.name === 'script-bottom';
+  if (isHeadOrBottomScript) {
```

Comment:
might need to exclude `External`s from this as well since they are loaded on-demand. (should be no issue as they don't need to be ssr-rendered / pre vue-compiled)

Reply:
> To double check -- do the plugins using _"inline"_ `<script>` tags still work? (disqus / copy code block / algolia)

Yup, I've tested those plugins and they seem to be working fine for me. 

However, I did notice one problem with Disqus plugin regarding SSR. Having `<disqus/>` will cause hydration issues because it is not a registered component for Vue. And as discussed before, any unrecognized HTML tags will cause hydration issues -- I will look into how I can fix this and make another PR to rectify it :-)

> might need to exclude Externals from this as well since they are loaded on-demand. (should be no issue as they don't need to be ssr-rendered / pre vue-compiled)

I've made use of `docId` to ensure that we will not process `External` scripts but I'm not sure if it's the best approach. I'm thinking if I can make a new parameter for the instantiation of `NodeProcessor`. This parameter will indicate where the `NodeProcessor` was instantiated (e.g. `Page`, `Layout`, or `External`) and we can use it to determine whether we should process the `<script>` tags. However, using `docId` would suffice actually. What do you think?

Also, just to make sure I'm understanding correctly, things would still work fine even if we process the `External` scripts, right? But just to ensure things are "done properly", we would not process the `External` scripts because they will not affect SSR (so no point processing them).

Reply:
> And as discussed before, any unrecognized HTML tags will cause hydration issues -- I will look into how I can fix this and make another PR to rectify it :-)

👍 

> This parameter will indicate where the NodeProcessor was instantiated (e.g. Page, Layout, or External) and we can use it to determine whether we should process the <script> tags. However, using docId would suffice actually. What do you think?

Since you're already passing the bucket `[]`, how about passing `undefined` in place of it and checking for that?

> Also, just to make sure I'm understanding correctly, things would still work fine even if we process the External scripts, right?

It would break for "externals" as the script / style tags simply disappear (collected but not used / hoisted)

e.g. a simple
```
<script>alert('hello world')</script>
```
would simply disappear from the external generated as `..._include_.html`.